### PR TITLE
feat: [CU-860rmfk15] create form to login

### PR DESCRIPTION
**Pull request by MASBERNATQ**
## Description

Create form to login.

## Changelog

- New translations
- Switch view between login/register in the `LoginModal`
- Refactor register form inside `LoginModal` in `RegisterForm` component
- Sign in to Supabase in the `LoginForm`
- Add _background-color_ behind eye toggle button in `FieldText` component

## Related ticket

This pull request is related to ClickUp ticket : **[CU-860rmfk15](https://app.clickup.com/t/860rmfk15)**.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have named the title of my pull request while respecting the naming convention:

```
<type>(<scope>): [<ticket id>] <short summary>
  │       │            │              │
  │       │            │              └─ Summary in present tense. Not capitalized. No period at the end.
  │       │            │
  │       │            └─ ClickUp ticket id (optional)
  │       │
  │       └─ Commit Scope (optional)
  │
  └─ Commit Type: build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test
```

### CODE REVIEW

**Comment by FlavianEng** [comment]
Task linked: [CU-860rmfk15 Create form to login](https://app.clickup.com/t/860rmfk15)

File: src/player/components/Login/LoginForm.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        if (error) {
+          setErrorSupabase(error.message);
+        } else {
+          resetForm();
+          setErrorSupabase('');
+          onClose();
+        }
+      })
+      .finally(() => setSubmitting(false));
+  };
```
Comment (FlavianEng) [inline src/player/components/Login/LoginForm.tsx:68]: That's reaaally weird do to an await with a then

File: src/player/components/Login/LoginForm.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        password: values.password,
+      })
+      .then(({ error }) => {
+        if (error) {
+          setErrorSupabase(error.message);
+        } else {
+          resetForm();
+          setErrorSupabase('');
+          onClose();
+        }
```
Comment (FlavianEng) [inline src/player/components/Login/LoginForm.tsx:65]: Please can you refactor this using the early return principle?

Doc : https://medium.com/swlh/return-early-pattern-3d18a41bba8#:~:text=R%20eturn%20early%20is%20the,when%20conditions%20are%20not%20met.

Comment (MASBERNATQ) [reply src/player/components/Login/LoginForm.tsx]: :heavy_check_mark:

File: src/player/components/Login/LoginForm.tsx (hunk 1 above)
Comment (MASBERNATQ) [reply src/player/components/Login/LoginForm.tsx]: :heavy_check_mark:

File: src/player/components/Login/LoginForm.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  });
+
+  /**
+   * Handle submit form.
+   *
+   * @param {PlayerLoginForm} values Form values
+   * @param {FormikHelpers<PlayerLoginForm>} formikHelpers Formik helpers
+   * @return {Promise<void>}
+   */
+  const handleSubmit = async (
```
Comment (FlavianEng) [inline src/player/components/Login/LoginForm.tsx:47]: Is this `async` still necessary?

Comment (MASBERNATQ) [reply src/player/components/Login/LoginForm.tsx:47]: yes comme le retour est une promesse

Comment (MASBERNATQ) [reply src/player/components/Login/LoginForm.tsx:47]: :heavy_check_mark:

**Review by FlavianEng — APPROVED** [review]

**Pull request merged**